infor-design/enterprise
# 1120 - Add Flex Toolbar to Header/Masthead/Formatters

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR adds a setting and some internal handling of HTML structure to some components that allow them to operate with Flex Toolbars in addition to standard Toolbars.  The following components are now capable of natively using Flex Toolbars:
- Header
- Masthead
- Editor

Additionally, some tests have been added to demonstrate the new features.  There's also been a Demoapp query parameter added that will transform most standard Toolbars in the examples into a Flex Toolbar for testing purposes (note that this is not yet available on all test pages).

**Related github/jira issue (required)**:
#1120 (doesn't close it just yet)

**Steps necessary to review your pull request (required)**:
Pull this branch and run the demoapp.  Then browse all the examples listed below, viewing the Flex Toolbar links, then removing the query parameter to view the standard Toolbar version, with the goal of having them line up closely.

_To test Header:_
- Navigate to http://localhost:4000/components/header/test-header-gauntlet.html? flextoolbar=true and make sure everything in the Header is displayed correctly.
- Look at the following navigation pattern tests and make sure everything is displayed correctly:
  - http://localhost:4000/patterns/navigation-breadcrumbs.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-breadcrumbs-alternate.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-drilldown.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-popupmenu.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-tabs?flextoolbar=true
  - http://localhost:4000/patterns/navigation-tabs-alternate.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-toolbar.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-wizard.html?flextoolbar=true
  - http://localhost:4000/patterns/navigation-wizard-alternate.html?flextoolbar=true
- Make sure to test all the links above at smaller breakpoints as well as the desktop breakpoint.
- Make sure the new functional test for headers passes.

_To Test Masthead:_
- Navigate to http://localhost:4000/patterns/masthead.html?flextoolbar=true, see the Masthead rendering properly.
- Test Masthead component samples to make sure they all work as expected:
  - http://localhost:4000/components/masthead/example-index.html?flextoolbar=true
  - http://localhost:4000/components/masthead/test-different-button-types.html?flextoolbar=true
  - http://localhost:4000/components/masthead/test-popups-on-each-side.html?flextoolbar=true
  - http://localhost:4000/components/masthead/test-tooltips-on-buttons.html?flextoolbar=true

_To Test Editor:_
- Navigate to http://localhost:4000/components/editor/example-flex-toolbar.html and make sure everything works as expected (also check breakpoint)
- Ensure the new functional test passes.

**included in this PR:**
- [x] Functional tests for the feature.
